public/chat_header_button.tsx

```diff
@@ -52,7 +53,10 @@ export const HeaderChatButton = (props: HeaderChatButtonProps) => {
   const [appId, setAppId] = useState<string>();
   const [conversationId, setConversationId] = useState<string>();
   const [title, setTitle] = useState<string>();
-  const [flyoutVisible, setFlyoutVisible] = useState(() => getChatbotOpenStatus());
+  const flyoutVisible = useObservable(props.flyoutVisible$, props.flyoutVisible$.getValue());
+  const setFlyoutVisible = useCallback((flag) => {
+    props.flyoutVisible$.next(flag);
```

Comment:
Nit: Do we need to add `props.flyoutVisible$` as a dependency?

Reply:
Oh, sure.